src/streamsync/ui_manager.py

```diff
@@ -0,0 +1,70 @@
+from typing import Union
+
+from streamsync.core import StreamsyncSession, session_manager
+from streamsync.ui_utils import (Component, SessionComponentTree, UIError,
+                                 current_parent_container)
+
+
+class StreamsyncUI:
+    """Provides mechanisms to manage and manipulate UI components within a
+    Streamsync session.
+
+    This class offers context managers and methods to dynamically create, find,
+    and organize UI components based on a structured component tree.
+    """
+
+    def __init__(self, session_id: str):
+        self.session: Union[StreamsyncSession, None] = session_manager.get_session(session_id)
+        if self.session is None:
+            raise RuntimeError('Invalid session passed to the UI manager')
```

Comment:
I'd prefer double quotes to be used consistently, but enforcing this via code reviews isn't good for anyone's sanity.

I'll create another issue to install `pylint`, which I'm sure will improve the quality of our code in other ways too.